### ISSUE TITLE
[0.20.1] Test file detection fails on Node 18.6.0 or higher

### ISSUE THREAD

**Issue by favna** [vitest-dev/vitest#1749]
### Describe the bug

Given the combination of Vitest 0.20.1 and Node of 18.6.0 or higher (18.6.0 or 18.7.0 at time of writing) test detection is failing. This problem does not occur with either Vitest 0.19.x, OR using Node 18.5.0. 

Error that is thrown:

<img width="1440" alt="image" src="https://user-images.githubusercontent.com/4019718/182021708-40218ad4-f26e-4c49-a8bb-30c45c692670.png">

<details><summary> Serialized error from screenshots </summary>

```
Serialized Error: {
  "errors": [
    [Error: "file:///Users/favna/workspace/favware/repro-repo/node_modules/vitest/dist/loader.mjs 'resolve'" did not call the next hook in its chain and did not explicitly signal a short circuit. If this is intentional, include `shortCircuit: true` in the hook's return.],
  ],
}
```

</details>

### Reproduction

I created a repository for reproduction: https://github.com/favna/friendly-dollop

1. Make sure you have NodeJS 18.6.0 or 18.7.0 installed
   - Alternatively, the repo has Node 18.7.0 pinned through [Volta](https://volta.sh)
2. Install dependencies with your package manager of choice
3. Run `vitest run` (i.e. through the `test` script)
4. Observe that no tests are detected
5. Make sure you have NodeJS 18.5.0 or lower installed
6. Run `vitest run`
7. Observe tests are detected

### System Info

```shell
System:
    OS: macOS 12.4
    CPU: (8) arm64 Apple M1
    Memory: 137.00 MB / 16.00 GB
    Shell: 5.8.1 - /bin/zsh
  Binaries:
    Node: 18.7.0 - /private/var/folders/jy/w45_1th97698mzgm_nsfl35w0000gn/T/xfs-661fa17a/node
    Yarn: 3.2.2 - /private/var/folders/jy/w45_1th97698mzgm_nsfl35w0000gn/T/xfs-661fa17a/yarn
    npm: 8.15.0 - ~/.volta/tools/image/node/18.7.0/bin/npm
  Browsers:
    Safari: 15.5
  npmPackages:
    vitest: ^0.20.1 => 0.20.1
```

### Used Package Manager

yarn

### Validations

- [X] Follow our [Code of Conduct](https://github.com/vitest-dev/vitest/blob/main/CODE_OF_CONDUCT.md)
- [X] Read the [Contributing Guidelines](https://github.com/vitest-dev/vitest/blob/main/CONTRIBUTING.md).
- [X] Read the [docs](https://vitest.dev/guide/).
- [X] Check that there isn't [already an issue](https://github.com/vitest-dev/vitest/issues) that reports the same bug to avoid creating a duplicate.
- [X] Check that this is a concrete bug. For Q&A open a [GitHub Discussion](https://github.com/vitest-dev/vitest/discussions) or join our [Discord Chat Server](https://chat.vitest.dev).
- [X] The provided reproduction is a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

**Comment by sheremet-va** [comment]
I see this error with Node 18.6:
```
"file:///Users/sheremet/local-git/friendly-dollop/node_modules/vitest/dist/loader.mjs 'resolve'" did not call the next hook in its chain and did not explicitly signal a short circuit. If this is intentional, include `shortCircuit: true` in the hook's return.
```

I guess this is the reason: https://github.com/nodejs/node/blob/main/doc/changelogs/CHANGELOG_V18.md#experimental-esm-loader-hooks-api

**Comment by favna** [comment]
Ah yes apologies I forgot to copy over the error but that is indeed the error I get also

**Comment by sheremet-va** [comment]
@favna should be fixed in `v0.20.2`, can you confirm?

**Comment by favna** [comment]
@sheremet-va Can confirm. Renovate created PRs to several repos and they passed, i.e. https://github.com/favware/colorette-spinner/pull/19
